x/poe/keeper/keeper.go

```diff
@@ -58,6 +58,14 @@ func (k *Keeper) SetPoEContractAddress(ctx sdk.Context, ctype types.PoEContractT
 	k.contractAddrCache.Store(ctype, contractAddr)
 }
 
+// InitContractAddressCache add all poe contracts to the in memory cache
+func (k *Keeper) InitContractAddressCache(ctx sdk.Context) {
+	k.IteratePoEContracts(ctx, func(contractType types.PoEContractType, address sdk.AccAddress) bool {
```

Comment:
I saw this was inserted in cache in `SetPoEContractAddress` and good addition to restore the cache on restart